### ISSUE TITLE
Access page's last modified date without setting front matter on every page

### ISSUE THREAD

**Issue by ckuehl** [11ty/eleventy#443]
I'm working on setting up a sitemap and would ideally like to access the page's last modified date versus its created date. I could override the date on each of my pages as documented at https://www.11ty.io/docs/dates/, but I'm curious if there's an easier way to access the last modified date for each item in the collection, solely for the purpose of the sitemap.

Thanks in advance!

**Comment by zachleat** [comment]
Ah, this is a good point. You’re familiar with directory data files, yeah? You should be able to set the date in one of those, no? https://www.11ty.io/docs/data-template-dir/

**Comment by Ryuno-Ki** [comment]
Hm, what if I only want to have that for the sitemap?

Imagine I am publishing a blog post.
At this point in time, the date is the same for my blog post and my sitemap (created date = last updated date).

Now imagine that I get slashdotted and edit the post with some remark some days later.

My expectation would be that the blog posts appears in the same order in the blog list (created date), but with the newer date in the sitemap (last updated).

Is that possible?
Or could I use [`fs.stat()`](https://nodejs.org/dist/latest-v11.x/docs/api/fs.html#fs_fs_stat_path_options_callback) to get the needed information myself in the sitemap renderer?

**Comment by robdodson** [comment]
I'd also really like to access this information. We'd like to separate a doc's created date from its last updated date so we can signal to folks when a doc is getting stale.

**Comment by zachleat** [comment]
> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by stedman** [comment]
Casting my vote for this feature add as well. 

I've been hacking something like the following into collections to get the file `created` and `modified` dates for use in sitemap.xml and feed.xml. I consider the post date to be a "publish" date (which has a midnight time stamp if derived from the file name) that may be different from the created date (which has a more accurate date+time stamp).

```js
const addFileDates = (posts) => posts.map((post) => {
  if (!(post && post.inputPath)) return post;

  const stat = fs.statSync(post.inputPath) || {};
  const newPost = post;

  newPost.dateCreated = stat.birthtime;
  newPost.dateModified = stat.mtime;

  return newPost;
});

module.exports = function (eleventyConfig) {
  eleventyConfig.addCollection('posts', function (collection) {
    const posts = collection.getFilteredByGlob('./_posts/**');

    return addFileDates(posts);
  });
};
```

**Comment by readingwebagency** [comment]
You can do this in your config file: eleventyConfig.addGlobalData("date", "Last Modified");

**Comment by zachleat** [comment]
The option described by @readingwebagency is a way to change the `page.date` default globally (or you set this via template data directory files to scope it to a specific directory).

The other option is to change the default behavior of `page.date` via more advanced configuration https://www.11ty.dev/docs/dates/#configuration-api-for-custom-date-parsing which shipped with v3.0.0

You could use the above to set your own special `date` value that can be handled with your own configuration code.

If you _don’t_ want to change the default behavior for `page.date` (since OP suggested this use case was specific to the sitemap), I’d suggest creating a filter to calculate it via `node:fs`’s `statSync` method using `page.inputPath`. See https://www.11ty.dev/docs/filters/#scoped-data-in-filters

I think these options are sufficient to satisfy the use case enough to take it off of the Enhancement Queue, but please weigh in if you disagree!
